code/game/gamemodes/survivor.dm

```diff
@@ -0,0 +1,460 @@
+    if(isnewplayer(M.current))
+        var/mob/new_player/N = M.current
+        N.close_spawn_windows()
+        N.new_character = X
+
+    M.transfer_to(X, TRUE)
+
+    to_chat(X, "<B>You are now an alien!</B>")
+    to_chat(X, "<B>Your job is to spread the hive and protect the Queen. If there's no Queen, you can become the Queen yourself by evolving into a drone.</B>")
+    to_chat(X, "Talk in Hivemind using <strong>;</strong> (e.g. ';My life for the queen!')")
+
+    X.update_icons()
+
+
+/datum/game_mode/survivor/proc/transform_queen(datum/mind/M)
+    // var/mob/living/carbon/Xenomorph/Queen/X = new (pick(GLOB.survivor_spawn_xeno))
+    var/mob/living/carbon/Xenomorph/Queen/X = new (pick(GLOB.xeno_spawn))
+
+    if(isnewplayer(M.current))
+        var/mob/new_player/N = M.current
+        N.close_spawn_windows()
+        N.new_character = X
+
+    M.transfer_to(X, TRUE)
+
+    to_chat(X, "<B>You are now the alien queen!</B>")
+    to_chat(X, "<B>Your job is to spread the hive.</B>")
+    to_chat(X, "Talk in Hivemind using <strong>;</strong> (e.g. ';My life for the hive!')")
+
+    X.update_icons()
+
+
+/datum/game_mode/survivor/proc/spawn_queen_in_ovi()
+    var/mob/living/carbon/Xenomorph/Queen/Q = new (pick(GLOB.xeno_spawn))
+    Q.ovipositor = TRUE
+    Q.away_time = 0 // queen so they can't be taken by late joins
+
+
+/datum/game_mode/survivor/proc/transform_survivor(datum/mind/M, late_join = FALSE,)
+    var/spawn_point = pick(GLOB.survivor_spawn_human)
+    if (late_join)
+        var/mob/A = pick(GLOB.alive_human_list)
+        if(A && !isspaceturf(get_turf(A.loc)))
+            spawn_point = A.loc
+
+    var/mob/living/carbon/human/H = new(spawn_point)
+
+    if(isnewplayer(M.current))
+        var/mob/new_player/N = M.current
+        N.close_spawn_windows()
+        N.new_character = H
+
+    M.transfer_to(H, TRUE)
+    H.client.prefs.copy_to(H)
+
+    var/job_title
+    var/role_guide
+    var/survivor_job = pick(subtypesof(/datum/job/survivor))
+
+    switch(rand(0, 100))
+        if (0 to 25)
+            survivor_job = /datum/job/survivor/atmos
+            job_title = SURVIVOR_BUILDER
+            role_guide = "You have extra building equipment, use the metal and tools to create defenses. You won't know you'll need it until its too late..."
+        if (25 to 50)
+            survivor_job = /datum/job/survivor/doctor
+            job_title = SURVIVOR_MEDIC
+            role_guide = "You have extra medical supplies, that could be useful if you or someone else gets hurt..."
+        if (50 to 75)
+            survivor_job = /datum/job/survivor/security
+            job_title = SURVIVOR_FIGHTER
+            role_guide = "You have found a weapon! Hope you don't have to use it too soon..."
+        if (75 to 100)
+            job_title = SURVIVOR_SCAV
+            role_guide = "You have a radio and pinpointer! Maybe this will lead you to what you need most..."
+
+    var/datum/job/J = new survivor_job
+
+    H.set_rank(J.title)
```

Comment:
AssignRole is better, set_rank is meant for admin-only interactions and thus is inefficient